### PR TITLE
initial commit of cloud tutorial notebook

### PR DESCRIPTION
**Pull request by andypbarrett**
Add tutorial notebook for UWG cloud demo.  This works on the 2i2c Openscapes instance

### CODE REVIEW

**Comment by jroebuck932** [comment]
@andypbarrett I didn't realize there was a pull request already open on this branch. Hopefully my commits are descriptive enough, but essentially I just updated the notebook to match our tutorial template and changed earthdata code to earthaccess. I haven't added a rendered version of the notebook yet, as I wanted to get your feedback/suggestions first. Thanks!

**Comment by andypbarrett** [comment]
I think the text, especially in the overview section, could be reworked and reordered.  There is quite a bit of repetition.  I think repetition is OK where it reinforces concepts but it is not always necessary.
 
I'm struggling with what the main objective of this notebook is.  On the one hand, it demonstrates using `earthaccess` to search for and access cloud-hosted data.  On the other hand, it demonstrates how to open ICESat-2 data in HDF5 format using `xarray`.  The plotting bit is an add-on to round out the workflow.  I think it would be helpful to highlight the generic nature of the search and access part.  We also need to convey that the user is in the cloud and the data is in the cloud.
 
I have three possible titles.
1. "Accessing and working data in the cloud: an example using ICESat-2."
2. "Accessing and working with ICESat-2 data in the cloud".
3. "Accessing and working with cloud-hosted ICESat-2 data in the cloud".

The first paragraph needs to be corrected.  I had originally put ATL03 but we actually use ATL06).  

A suggested first paragraph:

"This notebook demonstrates searching for _cloud-hosted_ ICESat-2 data and directly accessing a Land Ice Height (ATL06) file from an Amazon Compute Cloud (EC2) instance using the `earthaccess` package.   NASA data "in the cloud" are stored in AWS Simple Storage Service (S3) Buckets.  **Direct Access** is an efficient way to work with data stored in an S3 Bucket when you are working in the cloud.  Cloud-hosted files can be opened and loaded into memory without the need to download them first.  This allows you to take advantage of the scalability and power of cloud computing.

As an example dataset, we use ICESat-2 Land Ice Height (ATL06) over the Juneau Icefield, AK, for March 2003.  ICESat-2 data files, including ATL06, are stored in HDF5 format.  We demonstrate how to open an HDF5 and access data variables using `xarray`.  Land Ice Heights are then plotted using `hvplot`."

I have asked the cloud devs for a plain language definition of direct access.

I think the S3 token blurb can be moved to a section that describe `earthaccess` and why we use it.  Maybe all we need to say is that `earthaccess` manages getting S3 access tokens.

_on-prem_ is jargon.  I'd have to look at some earlier discussions but I think NSIDC-DAAC hosted vs cloud-hosted might be a good way to distinguish on-prem from cloud-hosted.

Add "an AWS EC2 instance in region us-west-2" to prerequisites.

Do we need the list of packages after the prerequisites?  I forget what the template is.  If we need to have packages listed than maybe remove them from learning objectives.  However, that seems a better place for them.

In Authenticate section: "Login requires **your** Earthdata login and password..."

I wonder if it makes more sense for the description/explanation of what the `get` method returns to be after the command.

Maybe we need a sub-heading for "Searching for cloud-hosted" data.

I can't decide if "spatiotemporal" is too jargony or not.  Is "search using a geographic bounding box and date range" better?

I think now that the ICESat-2 data is public `Query.hits()` will return the count of files found.

We had a discussion in Openscapes about whether we should use granules (NASA-speak) or files.  The consensus was to use files.  I'm not sure what USO does.  It is probably better to be consistent.

I talk about photon clouds in the markdown but the code deals with land ice heights.  I need to change this.

**Comment by jroebuck932** [comment]
@andypbarrett @asteiker this is ready for another review when you have time. Andy I have hopefully addressed all your comments. One thing I want to highlight is that I have ensured the terms granules and collections have been used throughout. I chose this instead of files and data sets because earthaccess uses the granules and collections terminology and I was trying to keep it consistent with that. I understand that openscapes have decided on files, and I did briefly raise this issue with Seamus and USO, but in short I think this needs a larger discussion about how and which terminology to use.

**Comment by asteiker** [comment]
@jroebuck932 Thanks for raising this issue! I know that @andypbarrett and I had gone back and forth on granule vs file, as there are pros/cons to using both terms. I think guidance from SCG would be really helpful here so that we can be consistent across all our content at NSIDC. Given that earthaccess utilizes "granule" and "collection", as these are the terms used on the backend within CMR, then I think it's reasonable to stick with those.

**Comment by andypbarrett** [comment]
I agree or granules vs files, etc.  Consistency is important.  I think a glossary would help to explain granules == files and collections == datasets.

**Comment by andypbarrett** [comment]
I think this looks good.  A few typos and suggestions.  I ran the notebook in 2i2c and had no problems.

Here are my comments, thoughts and suggestions.

Under Learning Objectives: suggest **"you"** will be able rather than "we".

Prerequisites: Suggest "So you need to create an EC2 instance in the us-west-2 region"  Do we have a glossary of cloud terms or an intro to the cloud that explains instances.  If not we should explain it here.

Import Packages: "This tutorial requires the following packages:"

"This can be entered using an Environment Variable - the default strategy - or using either a.netrc file or interactively."  I am wondering if this is necessary.  Could we just say "We use a .netrc strategy.  For other strategies see [`earthaccess.login`](https://nsidc.github.io/earthaccess/user-reference/api/api/#earthaccess.api.login)"

The code cell for Authentication uses "interactive" but we say we will use netrc.

Under Use Direct-Access to...  " using **a** package"

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:1]: We've kept our folder names fairly descriptive so that it gives some clue as to the data and/or tool demonstrated. How about changing to something like "ICESat-2_Cloud_Access"?

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:20]:
```suggestion
    "**Note: This is an updated version of the notebook that was presented to the NSIDC DAAC User Working Group in May 2022**\n",
```

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:28]:
```suggestion
    "`earthaccess` is a package developed by Luis Lopez (NSIDC developer) to allow easy search of the NASA Common Metadata Repository (CMR) and download of NASA data collections.  It can be used for programmatic search and access for both _DAAC-hosted_ and _cloud-hosted_ data. It manages authenticating using Earthdata Login credentials which are then used to obtain the S3 tokens that are needed for S3 direct access. https://github.com/nsidc/earthaccess\n",
```

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:33]:
```suggestion
    "The notebook was created by Andy Barrett, NSIDC, updated by Jennifer Roebuck, NSIDC, and is based on notebooks developed by Luis Lopez and Mikala Beig, NSIDC.\n",
```

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:47]:
```suggestion
    "1. We will be accessing data in Amazon Web Services (AWS) Simple Storage Service (S3) buckets. This requires an AWS account. For details on how to set one up see [here](https://nsidc.org/data/user-resources/help-center/nasa-earthdata-cloud-data-access-guide#anchor-1).\n",
```

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:49]: I think Andy may have already suggested this, but perhaps switch to "you" instead of "we" while describing the prerequisites. I think "we" is preferred while teaching live, but in this context it reads a bit funny.

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:51]: I like how 5. this is worded, starting with a statement describing what needs to happen in order for the notebook to run. Could we change the others to match this style? So something like:
1. An Amazon Web Services (AWS) account is created. We will be accessing data...
2. An EC2 Instance in us-west-2 is setup...
3. This one is already good to go, starting with "An Earthdata Login account..."!
4. A .netrc file is created in your home directory...

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:73]:
```suggestion
    "The first step in any `python` script or notebook is to import packages.  This tutorial requires the following pacakges:\n",
```

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:121]: Should we be using .netrc here instead since that is what we suggest above, and to be consistent with our other notebooks? I know we also need this for our regression tests to run, but this one is going to be an exception given that it must be running in us-west-2 (a future problem that we need to solve).

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:296]: Not sure if we want to leave this comment in about it not rendering so well?

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:321]:
```suggestion
    "Direct-access to data from an S3 bucket is a two step process.  First, the files are opened using the `open` method.  The `auth` object created at the start of the notebook is used to provide Earthdata Login authentication and AWS credentials.\n",
```

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:325]: Do we want to keep the '*' before "ICESat-2" (following "beam" above)? I think it is formatting to italicized because there is another '*' at the end of the sentence. 
```suggestion
    "*ICESat-2 measures photon returns from 3 beam pairs numbered 1, 2 and 3 that each consist of a left and a right beam"
```

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (asteiker) [inline notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:376]:
```suggestion
    "1. use `earthaccess` to search for ICESat-2 data using spatial and temporal filters and explore search results;\n",
```

**Review by asteiker — CHANGES_REQUESTED** [review]
Looks really good @jroebuck932! Just a few minor suggestions here.

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (jroebuck932) [reply notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:296]: Agreed, will remove

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (jroebuck932) [reply notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:51]: Agreed, will make these changes.

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (jroebuck932) [reply notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:49]: Agreed, will make these changes

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (jroebuck932) [reply notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb:1]: Agreed, will rename the folder to the suggested name

**Comment by jroebuck932** [comment]
@andypbarrett @asteiker have made all the suggested changes. Andy please take another quick look through, and if all looks good, then before merging I will create a rendered version of the notebook to add to the repo.

**Comment by andypbarrett** [comment]
The only item I see that needs to be changed is to remove the `import pprint` because it is not used, as far as I can see.  I'm happy to merge once that is done.

File: notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb
Comment (andypbarrett) [reply notebooks/DAAC_UWG_cloud_tutorial/nsidc_daac_uwg_cloud_access_tutorial.ipynb]: Agree.

**Comment by jroebuck932** [comment]
@andypbarrett thank you for making that change. Long story short I am having an issue creating an accurate rendered version of the notebook as there is a bug with the latest version of earthaccess  and the S3 access (`earthaccess.open`) doesn't work. So I am wondering if we want to wait until that is fixed in the new version, before we merge this? I checked with Luis and I think a new version should available next week.

In more detail, the S3 access does work with the previous version of earthaccess, but in the tutorial we state that `Query = earthaccess.collection_query().keyword('ICESat-2')` will list both cloud and ECS hosted data sets but with the previous version of earthaccess it will only list the cloud hosted data sets.

**Comment by asteiker** [comment]
> creating an accurate rendered version of the notebook

@jroebuck932 Is this just a problem on the rendered notebook? I'd rather this not get hung up on another release of earthaccess but I don't know if I fully understand the problem. Perhaps we could merge this without the rendered version first and come back and update with a rendered notebook after this is fixed.

**Comment by jroebuck932** [comment]
@asteiker @andypbarrett I have added Readme.md and environment.yml in the notebook folder and also updated the main README.md.  I think this is ready for merging, unless you want to take a quick look at the Readme files.

**Review by asteiker — APPROVED** [review]

**Comment by asteiker** [comment]
We will merge this with earthaccess v0.5.0 pinned, and will update once v0.5.2 is released, along with tool metadata for the nsidc website, updated main Readme, and we may also want to caveat usability in Binder given that this requires you to be running in-region.

**Pull request merged**
